cypress/e2e/audio-player/audio-player.spec.ts

```diff
@@ -0,0 +1,588 @@
+                    cy.contains(".mx_AudioPlayer_seek [role='timer']", "00:00").should("exist");
+
+                    // Assert that the play button is rendered
+                    cy.get("[data-testid='play-pause-button'][aria-label='Play']").should("exist");
+                });
+            });
+        });
+    });
+
+    it("should support downloading an audio file", () => {
+        visitRoom();
+
+        // Upload an audio file
+        uploadFile("cypress/fixtures/1sec.ogg");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            // Assert the audio player is rendered
+            cy.get(".mx_EventTile_last .mx_AudioPlayer_container").should("exist");
+
+            cy.get(".mx_EventTile_last")
+                .realHover()
+                .within(() => {
+                    // Click "Download" button on MessageActionBar
+                    cy.get('[aria-label="Download"]').click({ force: false });
+
+                    // Assert that the file was downloaded
+                    cy.readFile("cypress/downloads/1sec.ogg").should("exist");
+                });
+        });
+    });
+
+    it("should support replying to audio file with another audio file", () => {
+        visitRoom();
+
+        // Upload one second audio file with a long file name
+        uploadFile("cypress/fixtures/1sec-long-name-audio-file.ogg");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            // Assert the audio player is rendered
+            cy.get(".mx_EventTile_last .mx_AudioPlayer_container").should("exist");
+
+            cy.get(".mx_EventTile_last")
+                .realHover()
+                .within(() => {
+                    // Click "Reply" button on MessageActionBar
+                    cy.get('[aria-label="Reply"]').click({ force: false });
+                });
+        });
+
+        // Reply to the player with another audio file
+        uploadFile("cypress/fixtures/1sec.ogg");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            cy.get(".mx_EventTile_last").within(() => {
+                // Assert that replied audio file is rendered as file button inside ReplyChain
+                cy.get(".mx_ReplyChain_wrapper").within(() => {
+                    cy.get(".mx_MFileBody_info[role='button']").within(() => {
+                        // Assert that the file button has file name
+                        cy.get(".mx_MFileBody_info_filename").should("exist");
+                    });
+                });
+
+                cy.get(".mx_MAudioBody").within(() => {
+                    // Assert that the play button is visible
+                    cy.get("[data-testid='play-pause-button'][aria-label='Play']").should("be.visible");
+                });
+            });
+        });
+
+        // Scroll to the bottom to make the audio player visible for Percy tests
+        cy.get(".mx_MainSplit .mx_ScrollPanel").scrollTo("bottom");
```

Comment:
maybe move this earlier, and combine the last two `cy.get(".mx_RoomView_MessageList").within` calls?

(similarly in the "reply chain" test)